### PR TITLE
changes to readme

### LINKED ISSUE #221: Detail lacking on expected values for environment variables to enable testing
The readme.md file states that to run the unit tests in the azure-event-hubs-java project, we must set the following three environment variables:

1. EVENT_HUB_CONNECTION_STRING

1. PARTITION_COUNT

1. EPHTESTSTORAGE

What the docs do not make clear to a layperson such as myself is what the expected values are here. I am trying to build locally simply to get everything compiling under JDK 8 and JDK 9, as well as to work on any JavaDoc issues, and I'm blocked until I can get some further information. If you could clarify here or in the readme.md file, that would be much appreciated! Thanks.

### PR DESCRIPTION
**Pull request by SreeramGarlapati**
* cleanup readme & make samples more visible
* add description for Environment Variables for developers to run Unit Tests
* add eph to the readme

see the final result of readme.md @ https://github.com/SreeramGarlapati/azure-event-hubs-java/tree/finaltouch 

Close #221

### CODE REVIEW

File: Overview.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+
+# General Overview of Microsoft Azure Event Hubs client for Java
+
+This Java client library for Azure Event Hubs allows for both sending events to and receiving events from an Azure Event Hub. 
+
+An **event publisher** is a source of telemetry data, diagnostics information, usage logs, or other log data, as
```
Comment (sabeegrewal) [inline Overview.md]: We use publisher/publishing and sender/sending interchangeably. Could that be a source of confusion? Should we just stick to sender/sending?

File: Overview.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    ehClient.sendSync(sendEvent);
+```
+         
+The entire client API is built for Java 8's concurrent task model, generally returning 
+[*CompletableFuture<T>*](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html), so the library has these methods suffixed with *Sync* as their Synchronous counterparts/varaints.
+
+Learn more about publishing events, including advanced options, and when you should and shouldn't use those options, 
+[in the event publisher guide](PublishingEvents.md).
+
+### Consuming Events
```
Comment (sabeegrewal) [inline Overview.md]: Same goes for consumer/consuming and receiver/receiving. Not sure if it's overkill to stick to one word for each but I think that level of strictness will improve clarity in our docs.

File: Overview.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    final String sasKey = "---SharedAccessSignatureKey----";
+    ConnectionStringBuilder connStr = new ConnectionStringBuilder(namespaceName, eventHubName, sasKeyName, sasKey);
+		
+    EventHubClient ehClient = EventHubClient.createFromConnectionStringSync(connStr.toString());
+```
+
+Once you have the client in hands, you can package any arbitrary payload as a plain array of bytes and send it. 
+
+```Java
+    EventData sendEvent = new EventData(payloadBytes);
```
Comment (JamesBirdsall) [inline Overview.md]: EventData.create

File: Overview.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    ehClient.sendSync(sendEvent);
+```
+         
+The entire client API is built for Java 8's concurrent task model, generally returning 
+[*CompletableFuture<T>*](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html), so the library has these methods suffixed with *Sync* as their Synchronous counterparts/varaints.
```
Comment (JamesBirdsall) [inline Overview.md]: variants

File: Overview.md (hunk 1 above)
Comment (SreeramGarlapati) [reply Overview.md]: I agree with the feedback.
overview.md is a split from readme.md - not added newly. 
we eventually want to remove this file as - per the sdk fundamentals teams' guidelines - we need to de-dup our documentation from GitHub repos vs standard documentation channels.

hence, I am deferring the comments in overview.md file for now - to get the main issue of setting EnvironmentVariables for CITs merged.

**Comment by SreeramGarlapati** [comment]
@sabeegrewal / @JamesBirdsall 

I agree with the feedback on overview.md file. But, overview.md is a split from readme.md - not added newly :). 
we eventually want to remove this file as - per the sdk fundamentals teams' guidelines - we need to de-dup our documentation from GitHub repos vs standard documentation channels.

hence, I am deferring the comments in overview.md file for now - to get the main issue of setting EnvironmentVariables for CITs merged.

**Pull request merged**